### PR TITLE
cockpit.defer cleanup

### PR DESCRIPTION
**Pull request by martinpitt**
This doesn't fix all cockpit.defer()s yet. Some are *really* hard and need API work, and for some others I just ran out of time.

### CODE REVIEW

**Comment by jelly** [comment]
> This doesn't fix all cockpit.defer()s yet. Some are _really_ hard and need API work, and for some others I just ran out of time.

That's what I thought when looking at all the occurrences of `cockpit.defer` so this seems fine! :+1:

File: pkg/lib/serverTime.js [hunk 1]
```diff
@@ -106,7 +106,7 @@ export function ServerTime() {
     self.wait = function wait() {
         if (remote_offset === null)
             return self.update();
-        return cockpit.resolve();
+        return Promise.resolve();
```
Comment (cockpituous) [inline pkg/lib/serverTime.js:109]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/lib/service.js [hunk 2]
```diff
@@ -253,7 +254,7 @@ export function proxy(name, kind) {
     systemd_manager.addEventListener("JobRemoved", on_job_new_removed_refresh);
 
     function wait(callback) {
-        wait_callbacks.promise.then(callback);
+        wait_promise.then(callback);
```
Comment (cockpituous) [inline pkg/lib/service.js:257]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/interfaces.js [hunk 3]
```diff
@@ -1006,7 +1006,7 @@ export function NetworkManagerModel() {
                                               settings_to_nm(settings), objpath(this), objpath(specific_object))
                             .then(([path, active_connection]) => active_connection);
                 } catch (e) {
-                    return cockpit.reject(e);
+                    return Promise.reject(e);
```
Comment (cockpituous) [inline pkg/networkmanager/interfaces.js:1009]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/interfaces.js [hunk 4]
```diff
@@ -1397,7 +1397,7 @@
             return device.activate_with_settings(settings);
         } else {
             cockpit.warn("No way to apply settings", connection, settings);
-            return cockpit.resolve();
+            return Promise.resolve();
```
Comment (cockpituous) [inline pkg/networkmanager/interfaces.js:1400]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/networkmanager/interfaces.js [hunk 5]
```diff
@@ -1686,7 +1686,7 @@
             if (member_settings.connection.group == group_settings.connection.uuid ||
                 member_settings.connection.group == group_settings.connection.id ||
                 member_settings.connection.group == group_iface)
-                return cockpit.resolve();
+                return Promise.resolve();
```
Comment (cockpituous) [inline pkg/networkmanager/interfaces.js:1689]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/client.js [hunk 6]
```diff
@@ -443,24 +443,22 @@ function init_model(callback) {
 
     function enable_udisks_features() {
         if (!client.manager.valid)
-            return cockpit.resolve();
+            return Promise.resolve();
```
Comment (cockpituous) [inline pkg/storaged/client.js:446]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/client.js [hunk 7]
```diff
@@ -443,24 +443,22 @@
...
         if (!client.manager.EnableModules)
-            return cockpit.resolve();
+            return Promise.resolve();
```
Comment (cockpituous) [inline pkg/storaged/client.js:448]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/client.js [hunk 8]
```diff
@@ -443,24 +443,22 @@
...
+                return Promise.allSettled([client.manager_lvm2.wait(), client.manager_iscsi.wait()])
+                        .then(() => {
+                            client.features.lvm2 = client.manager_lvm2.valid;
+                            client.features.iscsi = (client.manager_iscsi.valid &&
+                                                            client.manager_iscsi.SessionsSupported !== false);
+                        });
             }, function(error) {
                 console.warn("Can't enable storaged modules", error.toString());
-                return cockpit.resolve();
+                return Promise.resolve();
```
Comment (cockpituous) [inline pkg/storaged/client.js:461]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/client.js [hunk 9]
```diff
@@ -475,21 +473,21 @@
             function (success) {
                 // hack here
                 client.features.legacy_vdo = success;
-                return cockpit.resolve();
+                return Promise.resolve();
             },
             function () {
-                return cockpit.resolve();
+                return Promise.resolve();
```
Comment (cockpituous) [inline pkg/storaged/client.js:479]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/crypto-keyslots.jsx [hunk 10]
```diff
@@ -56,7 +56,7 @@ function get_tang_adv(url) {
     return cockpit.spawn(["curl", "-sSf", url + "/adv"], { err: "message" })
             .then(JSON.parse)
             .catch(error => {
-                return cockpit.reject(error.toString().replace(/^curl: \([0-9]+\) /, ""));
+                return Promise.reject(error.toString().replace(/^curl: \([0-9]+\) /, ""));
```
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:59]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/iscsi-panel.jsx [hunk 11]
```diff
@@ -91,16 +89,14 @@ export class IscsiPanel extends React.Component {
                                             address: _("Unable to reach server")
                                         };
 
-                                    dfd.reject(error);
+                                    reject(error);
```
Comment (cockpituous) [inline pkg/storaged/iscsi-panel.jsx:92]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/iscsi-panel.jsx [hunk 12]
```diff
@@ -91,16 +89,14 @@
...
                                 });
 
                         progress_callback(null, function () {
                             cancelled = true;
-                            dfd.reject();
+                            reject();
```
Comment (cockpituous) [inline pkg/storaged/iscsi-panel.jsx:97]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/iscsi-panel.jsx [hunk 13]
```diff
@@ -144,7 +140,7 @@
                                     if (err.message.indexOf("authorization") != -1)
                                         iscsi_add_with_creds(discover_vals, vals);
                                     else
-                                        return cockpit.reject(err);
+                                        return Promise.reject(err);
```
Comment (cockpituous) [inline pkg/storaged/iscsi-panel.jsx:143]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/iscsi-panel.jsx [hunk 14]
```diff
@@ -171,7 +167,7 @@
                                             username: true, // makes it red without text below
                                             password: _("Invalid username or password")
                                         };
-                                    return cockpit.reject(err);
+                                    return Promise.reject(err);
```
Comment (cockpituous) [inline pkg/storaged/iscsi-panel.jsx:170]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/lvol-tabs.jsx [hunk 15]
```diff
@@ -107,15 +107,15 @@ function lvol_and_fsys_resize(client, lvol, size, offline, passphrase) {
             if (size - crypto_overhead > vdo.physical_size)
                 return vdo.grow_physical();
             else if (size - crypto_overhead < vdo.physical_size)
-                return cockpit.reject(_("VDO backing devices can not be made smaller"));
+                return Promise.reject(_("VDO backing devices can not be made smaller"));
```
Comment (cockpituous) [inline pkg/storaged/lvol-tabs.jsx:110]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/lvol-tabs.jsx [hunk 16]
```diff
@@ -107,15 +107,15 @@
...
             else
-                return cockpit.resolve();
+                return Promise.resolve();
```
Comment (cockpituous) [inline pkg/storaged/lvol-tabs.jsx:112]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/lvol-tabs.jsx [hunk 17]
```diff
@@ -107,15 +107,15 @@
...
         } else if (size < orig_size) {
             // This shouldn't happen.  But if it does, continuing is harmful, so we throw an error.
-            return cockpit.reject(_("Unrecognized data can not be made smaller here."));
+            return Promise.reject(_("Unrecognized data can not be made smaller here."));
```
Comment (cockpituous) [inline pkg/storaged/lvol-tabs.jsx:115]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/utils.js [hunk 18]
```diff
@@ -460,7 +460,7 @@ export function get_available_spaces(client) {
 export function prepare_available_spaces(client, spcs) {
     function prepare(spc) {
         if (spc.type == 'block')
-            return cockpit.resolve(spc.block.path);
+            return Promise.resolve(spc.block.path);
```
Comment (cockpituous) [inline pkg/storaged/utils.js:463]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18487-20230310-143650-f841503b-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]

**Pull request merged**